src/transformers/modeling_tf_utils.py

```diff
@@ -404,7 +404,8 @@ def input_processing(func, config, input_ids, **kwargs):
     signature = dict(inspect.signature(func).parameters)
     signature.pop("kwargs", None)
     signature.pop("self", None)
-    parameter_names = list(signature.keys())
+    parameter_names_list = list(signature.keys())
+    parameter_names = set(parameter_names_list)
```

Comment:
Is it possible for the signature to have duplicate keys, or is this just to make `if x in parameter_names` faster?

Reply:
Haha yeah, I went overboard with this one -- with the number of lookups we do per call, it is faster to create the set, but we're talking about microseconds (went on to check it with `timeit`). Clearly not worth adding code.

Reverting.